jwst/associations/lib/rules_level3.py

```diff
@@ -112,14 +114,194 @@ def __init__(self, *args, **kwargs):
             ),
             Constraint_Optical_Path(),
             Constraint_Target(),
+            DMSAttrConstraint(
+                name='exp_type',
+                sources=['exp_type'],
+                value=(
+                    'mir_lrs-fixedslit'
+                    '|mir_lrs_slitless'
```

Comment:
Should ``nis_soss`` be included in this list, because it's an equivalent mode to ``mir_lrs_slitless`` (slitless, but single target always located at the same place)?

Reply:
More specifically, `mir_lrs-slitless` does not need to be here, and the reasons `nis_soss` and `nrc_tsgrism` are not here, is that these are both "always time series", and handled by the `Asn_TSO` rule.

Reply:
Got it. However, at some time in the future when a TSO option/switch is added for all observing modes, it will be possible for things like ``nis_soss`` to be treated as non-TSO, so additional logic may be needed then.

Reply:
So, the plan is to allow "always TSO" to also be switchable? I had understood that even when the option is there, that the "always TSO" one would not be allowed. Or should we just "CYA" this and plan for both?

Note: currently APT 26.0.3 allows addition of TSO special requirement, but it does not allow one to "take it away".

Reply:
Currently, nis_soss observations can be non-TSO. Should be treated as  TSO only when the flag says it is TSO. If the "Time Series Observation" is NOT present in the special requirements then it means it is a non-TSO. I believe that in one of the test cases I created for one of the OTB or EMTB tests I removed this flag. Though I don't remember if it got executed. So I agree with Howard that additional logic is needed.

Reply:
Clicking through APT 26.0.3 shows that TSO status for various modes is:
 - ``mir_image``: TSO default Off, but can be turned On
 - ``mir_lrs-slitless``: TOS default On, but can be turned Off
 - ``mir_lrs-fixedslit``: TSO not allowed
 - ``mir_mrs``:  TSO not allowed (but supposedly will be in future)
 - ``nrc_image``: TSO not allowed (use ``nrc_tsimage`` instead)
 - ``nrc_tsimage``: TSO required (cannot be turned Off)
 - ``nrc_tsgrism``: TSO required (cannot be turned Off)
 - ``nrc_wfss``: TSO not allowed (use ``nrc_tsgrism`` instead)
 - ``nrs_fixedslit``: TSO not allowed (use ``nrs_brightobj`` instead)
 - ``nrs_ifu``: TSO not allowed (but supposedly will be in future)
 - ``nrs_msaspec``: TSO not allowed
 - ``nrs_brightobj``: TSO required (cannot be turned Off)
 - ``nis_image``: TSO not allowed (but supposedly will be in future)
 - ``nis_wfss``: TSO not allowed (use ``nrc_tsgrism`` instead)
 - ``nis_soss``: TSO default On, but can be turned Off
 - ``nis_ami``: TSO not allowed

So the permanent TSO modes are ``nrc_tsimage``, ``nrc_tsgrism``, and ``nrs_brightobj``, while ``mir_image``, ``mir_lrs-slitless``, and ``nis_soss`` can be toggled on/off.